src/ignore_path.rs

```diff
@@ -11,7 +11,12 @@ impl IgnorePathSet {
         let mut ignore_builder = gitignore::GitignoreBuilder::new(ignore_list.rustfmt_toml_path());
 
         for ignore_path in ignore_list {
-            ignore_builder.add_line(None, ignore_path.to_str().unwrap())?;
+            let ignore_line = ignore_path.to_str().unwrap();
+            if cfg!(windows) {
+                ignore_builder.add_line(None, &ignore_line.replace("\\", "/"))?;
```

Comment:
Seems like one downside of only doing this on Windows is that projects utilizing `ignore` with these windows style paths could end up with different experiences with rustfmt on different platforms. 

A `cargo fmt -- --check` could pass for a developer/CI engine on Windows but fail for a developer/CI engine on Mac, Linux, etc.

Reply:
You are right, this is a real concern. However, it seems that the `#[path = "..."]` attribute does not support Windows-style paths in other environments as well:

```rust
// error: couldn't read foo\bar.rs: No such file or directory (os error 2)
#[path = "foo\\bar.rs"]
mod foobar;
```

I will just take this behavior an excuse for rustfmt that it's ok not to support the Windows-style path outside of Windows :wink: 

While you are this, could you add some documentation in `Configurations.md` about this, please?

Reply:
> I will just take this behavior an excuse for rustfmt that it's ok not to support the Windows-style path outside of Windows 😉

In that case, do you think that Windows-style paths should not be supported in rustfmt's `ignore` setting as well? Using regular `/` style paths in rustfmt `ignore` config works fine for me on both my Ubuntu and Windows machines, so I'd personally prefer to just use the `/` style that works on all platforms.

Instead of making this code change, could we just update the Configurations information to instruct users to use `/` paths for `ignore` setting?

Reply:
Now that you mention it, I think that sounds more reasonable.

> Instead of making this code change, could we just update the Configurations information to instruct users to use / paths for ignore setting?

Would you mind updating this PR with this direction? Sorry for multiple requests for changes, this should be the last one.

Reply:
No worries!